### PR TITLE
Append the ELF flags for sram-ecc and xnack to the target triple per code object

### PR DESCRIPTION
**Pull request by david-salinas**
…code object

### CODE REVIEW

File: include/hip/hcc_detail/code_object_bundle.hpp [hunk 1]
```diff
@@ -124,6 +126,18 @@ class Bundled_code_header {
...
+		std::stringstream tmp{std::string{y.blob.cbegin(), y.blob.cend()}};
+                ELFIO::elfio reader;
+                if (reader.load(tmp)) {
+			unsigned int xnack_enabled = reader.get_flags() & 0x100;
+			unsigned int sram_ecc_enabled = reader.get_flags() & 0x200;
+			if (xnack_enabled)
+				y.triple.append("+xnack");
+			if (sram_ecc_enabled)
+				y.triple.append("+sram-ecc");
+		}
```
Comment (scchan) [inline include/hip/hcc_detail/code_object_bundle.hpp]: indentation of this new block of code is off

Comment (david-salinas) [reply include/hip/hcc_detail/code_object_bundle.hpp]: done.

Comment (scchan) [reply include/hip/hcc_detail/code_object_bundle.hpp]: nope, line 130 is still misaligned and the number of space characters for the indent is different to other places in this same function

**Comment by deven-amd** [comment]
@mangupta 

I think this change will break the TF build.

I (accidentally) tried to do a TF build with the latest HIP (which includes this merge commit) and ran into the following error:

```
Execution platform: @bazel_tools//platforms:host_platform
In file included from bazel-out/k8-opt/genfiles/external/local_config_rocm/rocm/rocm/include/hip/hcc_detail/elfio/elfio.hpp:46:0,
                 from bazel-out/k8-opt/genfiles/external/local_config_rocm/rocm/rocm/include/hip/hcc_detail/code_object_bundle.hpp:35,
                 from bazel-out/k8-opt/genfiles/external/local_config_rocm/rocm/rocm/include/hip/hcc_detail/program_state.hpp:25,
                 from bazel-out/k8-opt/genfiles/external/local_config_rocm/rocm/rocm/include/hip/hcc_detail/hip_runtime_api.h:50,
                 from bazel-out/k8-opt/genfiles/external/local_config_rocm/rocm/rocm/include/hip/hip_runtime_api.h:308,
                 from bazel-out/k8-opt/genfiles/external/local_config_rocm/rocm/rocm/include/hip/hip_runtime.h:64,
                 from external/eigen_archive/unsupported/Eigen/CXX11/Tensor:87,
                 from ./third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1,
                 from ./tensorflow/core/kernels/scatter_nd_op.h:19,
                 from tensorflow/core/kernels/scatter_nd_op.cc:23:
bazel-out/k8-opt/genfiles/external/local_config_rocm/rocm/rocm/include/hip/hcc_detail/elfio/elfio_segment.hpp: In instantiation of 'void ELFIO::segment_impl<T>::load(std::istream&, std::streampos) [with T = ELFIO::Elf32_Phdr; std::istream = std::basic_istream<char>; std::streampos = std::fpos<__mbstate_t>]':
tensorflow/core/kernels/scatter_nd_op.cc:783:1:   required from here
bazel-out/k8-opt/genfiles/external/local_config_rocm/rocm/rocm/include/hip/hcc_detail/elfio/elfio_segment.hpp:149:15: error: exception handling disabled, use -fexceptions to enable
             } catch (const std::bad_alloc&) {
               ^
Target //tensorflow/python/kernel_tests:pooling_ops_test_gpu failed to build
```

I think for TF, the include for `elfio/elfio.hpp` needs to be done, the same way it is done in `program_state.hpp`

**Comment by jeffdaily** [comment]
If it is not appropriate to have a similar solution to `program_state.hpp`, perhaps the work-around would be as described here: https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_exceptions.html.  
Specifically the section https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_exceptions.html#intro.using.exception.no.

**Pull request merged**